### PR TITLE
Serve information about the host itself

### PR DESCRIPTION
**Pull request by tobikris**
I thought it would be a good idea to also serve records for the host itself.
You can configure the local endpoint ip in the Corefile optionally. If it is not set the plugin will try to determine it automatically (default outgoing ip). The port is based on the wgctrl information.

### CODE REVIEW

**Comment by jwhited** [comment]
interesting - what's the use case for serving the "server" info?

**Comment by tobikris** [comment]
I am using this to connect a mesh of multiple servers and am using public DNS. This allows for connecting servers that only know each others public key. This is great for my setup which is running on OpenStack with public IPs that are changing now and then. I just thought that this is not hurting and opens up some more usecases.

**Comment by jwhited** [comment]
Maybe the use via public DNS is where I'm getting lost, but something in that chain is aware of the static address of the "server", can that not be utilized?

Would love to understand your use case better.

File: setup.go [hunk 1]
```diff
@@ -29,6 +30,14 @@ func setup(c *caddy.Controller) error {
 	}
 	device := c.Val()
 
+	// parse optional local ip
+	var wgIP net.IP
+	if c.NextArg() {
+		wgIP = net.ParseIP(c.Val())
+	} else {
+		wgIP = getOutboundIP()
```
Comment (jwhited) [inline setup.go:38]: I think we can use `w.LocalAddr()` + `net.SplitHostPort` to get the local address for the socket handing the DNS request

File: wgsd.go [hunk 2]
```diff
@@ -63,19 +70,26 @@ func (p *WGSD) ServeDNS(ctx context.Context, w dns.ResponseWriter,
...
-	if len(device.Peers) == 0 {
-		return nxDomain(p.zone, w, r)
-	}
 
 	// setup our reply message
 	m := new(dns.Msg)
 	m.SetReply(r)
 	m.Authoritative = true
 
+	allPeers := []host{
```
Comment (jwhited) [inline wgsd.go:79]: Maybe call this type `peer` or use wgtypes.Peer and construct one for the CoreDNS host.

Will need to include AllowedIPs, which we can set to addresses obtained from the WireGuard interface.

**Comment by jwhited** [comment]
Hi @tobikris - Thanks for the PR. I added #18 to track this and left some feedback here.

If you want to rebase I'm happy to review and get this merged or open a new PR, just let me know.

**Comment by jwhited** [comment]
closed in #21

**Comment by tobikris** [comment]
Sorry that I was unresponsive for so long. I think my use-case is solved with #21. Thanks for the brilliant plugin!

**Pull request closed without merging**